lib/terrier/frontend/model_generator.rb

```diff
@@ -368,9 +370,10 @@ def compute_ref_type(ref)
     else
       t
     end
-    if ref.class == ActiveRecord::Reflection::HasManyReflection
-      t = "#{t}[]"
-    end
+    is_array_type = ref.is_a?(ActiveRecord::Reflection::HasManyReflection) ||
+                    (ref.is_a?(ActiveRecord::Reflection::ThroughReflection))
+
+    t += "[]" if is_array_type
```

Comment:
Actually I think it would be better to completely ignore `through` reflections. I don't think the Db query methods support those types of reflections, do they?

Reply:
Oh, huh. Alright

Reply:
In that case we should account for the possibility that it's a `has_one` with a through rather than a `has_many` with a through clause. See for example `LocationSale`'s `program` relation in clypboard.